File: examples/datasets/celestrak_download.py
# /// script
# dependencies = ["brahe"]
# ///
# FLAGS = [IGNORE]
"""
Download and save CelesTrak ephemeris data.

Demonstrates:
- Downloading ephemeris to a file
"""

import brahe as bh

if __name__ == "__main__":
    print("Download CelesTrak Ephemeris")
    print("=" * 60)

    # Download and save ephemeris
    output_file = "/tmp/gnss_satellites.json"
    bh.datasets.celestrak.download_ephemeris(
        "gnss", output_file, content_format="3le", file_format="json"
    )
    print(f"\nSaved GNSS ephemeris to: {output_file}")

    print("\n" + "=" * 60)

File: examples/datasets/celestrak_download.rs
//! FLAGS = [IGNORE]
//! Download and save CelesTrak ephemeris data.
//!
//! Demonstrates:
//! - Downloading ephemeris to a file

use brahe::datasets::celestrak;

fn main() {
    println!("Download CelesTrak Ephemeris");
    println!("{}", "=".repeat(60));

    // Download and save ephemeris
    let output_file = "/tmp/gnss_satellites.json";
    celestrak::download_ephemeris(
        "gnss",
        output_file,
        "3le",
        "json",
    )
    .unwrap();
    println!("\nSaved GNSS ephemeris to: {}", output_file);

    println!("\n{}", "=".repeat(60));
}
